src/components/structures/RoomView.js

```diff
@@ -159,6 +159,7 @@ module.exports = React.createClass({
         MatrixClientPeg.get().on("Room.name", this.onRoomName);
         MatrixClientPeg.get().on("Room.accountData", this.onRoomAccountData);
         MatrixClientPeg.get().on("RoomState.members", this.onRoomStateMember);
+        MatrixClientPeg.get().on("RoomMember.membership", this.onRoomMemberMembership);
```

Comment:
You could alternatively use `onRoomStateMember` instead, but I don't really care either way (1 extra listener isn't a big deal).